File: javascript/practice/selectionSort.js
function selectionSort(a){
  let minIndex = 99999;
  let n = a.length;
  for(let i = 0; i < n - 1; i++){
    minIndex = i;
    //find index of minimum value from remaining array (select min value in each loop)
    for(let j = i + 1; j < n; j++){
      if(a[j] < a[minIndex]){
        minIndex = j;
      }
    }
    //swap min index value with ith value
    let t = a[i];
    a[i] = a[minIndex];
    a[minIndex] = t;
  }
  return a;
}

console.log(JSON.stringify(selectionSort([3,2,1,4,6,5])));

File: javascript/practice/selection_sort.rb
def selection_sort(a)
  n = a.length
  i = 0
  while i < n - 1
    min = i
    j = i + 1
    while j < n
      if a[j] < a[min]
        min = j
      end
      j += 1
    end
    a[i], a[min] = a[min], a[i]
    i += 1
  end
  a
end

print selection_sort([3, 2, 1, 4, 6, 5])
puts
